### PR TITLE
Fix #33, remove frame-ancestors from Content-Security-Policy headers

### LINKED ISSUE #33: Content-Security-Policy can block pages
Use can use CSP to block framing a site. testpilot.firefox.com is an example, with `frame-ancestors 'self'`

Unfortunately modifying the request via the add-on does not affect the CSP :(

Note we can definitely *read* the CSP header, and detect that there will be a problem.

### PR DESCRIPTION
**Pull request by ianb**

### CODE REVIEW

**Comment by fzzzy** [comment]
There's a little lint problem.

File: addon/background.js [hunk 1]
```diff
@@ -113,14 +113,24 @@ browser.webRequest.onBeforeSendHeaders.addListener(function (info) {
...
   for (let i = 0; i < headers.length; i++) {
     let name = headers[i].name.toLowerCase();
     if (name === "x-frame-options" || name === "frame-options") {
       headers.splice(i, 1);
-      return {"responseHeaders": headers};
+      i--;
+      madeChanges = true;
+    }
+    if (name === "content-security-policy") {
+      headers[i].value = headers[i].value.replace(/frame-ancestors[^;]*/i, "");
```
Comment (fzzzy) [inline addon/background.js]: Just so I am sure I understand, this matches anything that starts with frame-ancestors, and then any characters up to a ;

Won't this leave the ; in the string? I guess that it is probably legal to have multiple ; in a row.

File: addon/background.js [hunk 2]
```diff
@@ -113,14 +113,24 @@ browser.webRequest.onBeforeSendHeaders.addListener(function (info) {
...
+      madeChanges = true;
     }
   }
-  return {};
+  if (madeChanges) {
+    return {"responseHeaders": headers};
+  } else {
+    return {};
```
Comment (pdehaan) [inline addon/background.js]: From the Circle-CI logs:

```
/home/circleci/project/addon/background.js
  131:10  error    Unnecessary 'else' after 'return'  no-else-return
```

I think it's saying we can just simplify to the following:

```js
if (madeChanges) {
  return {"responseHeaders": headers};
}
return {};
```

File: addon/background.js (hunk 1 above)
Comment (ianb) [reply addon/background.js]: It would, yeah. Since I have to fix a lint thing I'll just add `;?` here too

**Comment by ianb** [comment]
I'm going to read the comments as implicit approval (now that the lint errors are gone)

**Pull request merged**